src/coreclr/jit/fgbasic.cpp

```diff
@@ -368,6 +368,15 @@ void Compiler::fgRemoveEhfSuccessor(BasicBlock* block, const unsigned succIndex)
                   (succCount - succIndex - 1) * sizeof(FlowEdge*));
     }
 
+    // Recompute the likelihoods of the block's other successor edges.
+    const weight_t removedLikelihood = succEdge->getLikelihood();
+    for (unsigned i = 0; (removedLikelihood != 1.0) && (i < (succCount - 1)); i++)
```

Comment:
We still need to handle the case where removed likelihood is 1.0, don't we (probably rare)?

In that case we should just spread the likelihood equally.

Reply:
Right, fixed